streaming/base/dataset.py

```diff
@@ -709,27 +834,22 @@ def __getitem__(self, sample_id: int) -> Any:
         Returns:
             Dict[str, Any]: Mapping of column name to column data.
         """
-        # Locate the shard and sample offset within that shard where the sample lives.
         shard_id, shard_sample_id = self.index.find_sample(sample_id)
         shard = self.shards[shard_id]
-
         try:
-            # Attempt to directly access the sample for performance reasons.
+            # Shortcut path: just assume the shard is present. Manually update last access time
+            # afterward. Using exceptions as control flow is actually faster than doing it
+            # properly because python.
             sample = shard[shard_sample_id]
+            self._shard_access_times[shard_id] = time()
         except:
-            # Get handles to the shared shard states array and its protective file lock.
-            lock, shard_states = self._get_shard_states()
-
-            # Download the shard if not already being downloaded. Block if download in progress.
-            self._download_or_skip_shard(lock, shard_states, shard_id, True)
-
-            # Finally, access the sample.
+            # Proper path: shard wasn't present. Download the shard or wait for its download to
```

Comment:
add to comment that download shard sets the access time, i was confused for a sec why the try sets the time and the except doesn't

Reply:
done